tests/trainer/test_fsdp_param_groups.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+import gc
+
+import pytest
+import torch
+from packaging import version
+from torch.utils.data import DataLoader
+
+from composer.trainer.trainer import Trainer
+from composer.utils import dist, misc
+from tests.common import RandomClassificationDataset, SimpleModel, device, world_size
+
+
+@pytest.mark.parametrize('mixed_precision', ['DEFAULT'])
+@pytest.mark.parametrize('reentrant', [True, False])
+@pytest.mark.filterwarnings('ignore::UserWarning')
+@device('gpu')
+@world_size(1, 2)
```

Comment:
For FSDP tests, I think we should really just test with `world_size=1`?

Reply:
changed to run on world size 2 only!